Experimental/CameraComponent/Viewport.cs

```diff
@@ -0,0 +1,322 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Networking.Sockets;
+using Windows.Storage;
+using Windows.UI.Composition;
+using Windows.UI.Composition.Scenes;
+using Windows.UI.Xaml;
+
+namespace CameraComponent
+{
+    public enum Stretch { Fill, FixX, FixY, Uniform, UniformToFill };
+
+    public sealed class Viewport
+    {
+        private SceneVisual sceneVisual;
+        private Vector2 size;
+        private Vector3 offset;
+        private Stretch stretch;
+        private Camera camera;
+
+        private CompositionPropertySet propertySet;
+
+        public Viewport(SceneVisual visual)
+        {
+            sceneVisual = visual;
+            var windowSize = Window.Current.Bounds;
+            size = new Vector2((float)windowSize.Width, (float)windowSize.Height);

```

Comment:
I don't think we should initialize to the Window's bounds.  Maybe (0,0) or maybe something reasonable like (100,100).